lib/Tumblr/StreamBuilder/Codec/BinaryCodec.php

```diff
@@ -103,9 +103,15 @@ public function __construct(
     /**
      * @inheritDoc
      */
-    public function encode(Templatable $obj): string
+    public function encode($obj): string
     {
-        $json = Helpers::json_encode($obj->to_template());
+        if ($obj instanceof Templatable) {
+            $json = Helpers::json_encode($obj->to_template());
+        } elseif (is_string($obj)) {
+            $json = $obj;
+        } else {
+            throw new \InvalidArgumentException(sprintf("%s type is not supported", get_class($obj)));
```

Comment:
we should document this in the method's phpdoc, since now it doesn't match the parent class